Nil check on oncall integration templates

julienduchesne opened the pull request:
If a user defines an empty templates block, it currently panics 
An empty block should be functionally equivalent to not defining the block at all

julienduchesne committed:
Nil check on oncall integration templates
If a user defines an empty templates block, it currently panics
An empty block should be functionally equivalent to not defining the block at all

julienduchesne committed:
Set an empty templates in the tests

inkel reviewed (approved):
LGTM but there are some comments from a CI check 🤔

julienduchesne committed:
oops

julienduchesne committed:
Add test for all three cases
- no templates block
- empty templates block
- non-empty templates block

+ fix all cases

julienduchesne commented:
Asking for a new review. I added more test cases and a DiffSupressFunc (if the templates is empty or missing should be the same outcome)

inkel reviewed (approved):
LGTM!!

quatauta commented:
@julienduchesne Thank you so much for this PR! This improves our life with the Grafana Terraform Provider! 🎉
